### ISSUE TITLE
[Bug]: LC0096 triggers when function is also used without Rec

### ISSUE THREAD

**Issue by DominicSaladin** [ALCops/Analyzers#323]
### Analyzer

LinterCop

### Issue type

False positive (diagnostic fires when it shouldn't)

### Rule ID

LC0096

### Description

I have following procedure in a custom table:
```AL
local procedure GetSubtreeSize(ParentRibM13DeLine: Record "RIB M13 DE Line"): Integer
```

I call this function on multiple occasions:
<img width="308" height="187" alt="Image" src="https://github.com/user-attachments/assets/571f4c85-8fd1-4b0d-869f-4383d3b2c076" />

I use the function for different variable, some of them are `Rec`, but I'm not sure how I should fix this issue. I could introduce a overload that has no parameter, but this would still need to call the main function with `Rec` which would trigger the same thing again.

Any suggestion to how to fix this?

### AL code to reproduce

```al
local procedure Test()
var
    RibM13DeLine: Record "RIB M13 DE Line";
begin
    // triggers it
    GetSubtreeSize(Rec);

    // doesn't trigger it
    GetSubtreeSize(RibM13DeLine);
end;

local procedure GetSubtreeSize(ParentRibM13DeLine: Record "RIB M13 DE Line"): Integer
begin
    exit(1);
end;
```

**Comment by Arthurvdv** [comment]
To be honest, I'm not entirely sure what the best way to resolve this would be.

I'm going to tag @iwanscheidsnc for his input, as he originally created this rule in BusinessCentral.LinterCop and may have more insight into the reasoning behind it and how this scenario should be handled.

@iwanscheidsnc, would you mind sharing your thoughts on this?

**Comment by iwanscheidsnc** [comment]
The original intent behind this rule is that when you pass Rec as a parameter to a function, it should ideally be refactored so the function is called on Rec instead — i.e. Rec.GetSubtreeSize() and RibM13DeLine.GetSubtreeSize(). However, in this specific case that's not straightforward since the function is local and there's no good reason to make it public or internal just to satisfy that pattern.

I see a few options:

Option 1 – Add a parameterless overload:
Add a GetSubtreeSize() procedure without parameters. Inside, assign RibM13DeLine := Rec and then call GetSubtreeSize(ParentRibM13DeLine). It's slightly unfortunate to introduce an extra variable for this, but it satisfies the rule's intent.

Option 2 – Suppress with a #pragma warning disable
Treat this as a justified special case and suppress the diagnostic locally.

I think the rule itself is correct as-is. The situation where you genuinely want to call a local function using different record instances — including Rec — is a legitimate edge case, and the developer has  clean ways to handle it. I wouldn't classify this as a false positive that warrants changing the rule's behavior.

It could be considered to relax the rule for local procedures in tables where the Record parameter is passed without var. In that case the record is copied (call by value), so there is no actual mutation risk on Rec — the most important concern the rule guards against is less relevant. This might be a worthwhile refinement to reduce noise in such scenarios. But even without mutation risk, it is cleaner to avoid implicitly passing the same record twice — making the intent explicit improves readability and avoids subtle confusion at the call site.

**Comment by Arthurvdv** [comment]
@DominicSaladin and @iwanscheidsnc I've taken the liberty to updated the docs for this rule based on the info here. If you have additions or improvements feel free to create an PR to improve this further.
